src/SmartNumber.php

```diff
@@ -41,13 +46,13 @@ final class SmartNumber
 	/** @var string */
 	private $string;
 
-	/** @var string */
+	/** @var BigInteger */
```

Comment:
What is the limitation of `BigInteger` instead of `string`?

Reply:
You can store only arbitrary length integers in BigInteger, so it's safer.